### PR TITLE
[TACHYON-1119] Improve JsonJournalFormatterTest

### PR DESCRIPTION
**Pull request by saveveltri**
https://tachyon.atlassian.net/browse/TACHYON-1119

Hi everyone,
    this PR is for improving JsonJournalFormatterTest class. 
Every journal entries are tested against a json representation (in file JournalEntries.json).

A json file containing all the entries representation is provided and in class JournalFormatterTestBase a map of JournalEntry objects has been declared; therefore we have entries object samples and entries json samples
Using this data, the method entryJsonTest does most of the work. 
Given a journal entry type, a raw json is written into the journal and the related object is created by deserialize that json.
The test ensures that the entry obtained by json deserialization is the same as the related sample object.
In this way it is possible to check against unwanted json representation modifications.

Bests

### CODE REVIEW

**Comment by haoyuan** [comment]
Jenkins, test this please.

**Comment by haoyuan** [comment]
Jenkins, add to whitelist.

**Comment by saveveltri** [comment]
At first sight, it does not seem related to my PR.
I have worked only on the test "side".
At least, they would have been my tests to fail

**Comment by gpang** [comment]
Jenkins, test this please

**Comment by gpang** [comment]
Jenkins, test this please

File: servers/src/test/java/tachyon/master/journal/JsonJournalFormatterTest.java [hunk 1]
```diff
@@ -15,9 +15,119 @@
...
+    JsonNode n = sRootNode.get(type.toString());
+    new ObjectMapper().writeValue(mOs, n);
+    JournalEntry readEntry = read();
+    assertSameEntry(mDataSet.get(type), readEntry);
+  }
+
+  // Block
+
+  @Test
+  public void blockIdGeneratorEntryJsonTest() throws IOException {
```
Comment (yupeng9) [inline servers/src/test/java/tachyon/master/journal/JsonJournalFormatterTest.java]: Other than enumerating all the values one function per type, what about iterating over the key set of JournalFormatterTestBase's dataset?

File: servers/src/test/java/tachyon/master/journal/JournalFormatterTestBase.java [hunk 2]
```diff
@@ -77,6 +79,41 @@
   protected OutputStream mOs;
   protected InputStream mIs;
 
+  //map that holds test journal entries
+  protected Map<JournalEntryType, JournalEntry> mDataSet =
+      ImmutableMap.<JournalEntryType, JournalEntry>builder()
```
Comment (yupeng9) [inline servers/src/test/java/tachyon/master/journal/JournalFormatterTestBase.java]: Also shall we add a test for checking the size of `mDataset` is equal to the size of JournalEntryType's values? That will require the unit test for any further added Journal entry.

**Comment by yupeng9** [comment]
Thanks for improving the Journal format testing. I left a few comments.

**Comment by saveveltri** [comment]
Thanks for your suggestions.
I've updated my code in order to have all tests inside a loop.
I've added all the missing entries except for FREE, because I've noticed it is never used in the project and I've decided to comment it.
Please let me know if everything is fine.

File: servers/src/test/resources/JournalEntries.json [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+    "mSequenceNumber":1945,
+    "mType":"REQUEST_FILE_PERSISTENCE",
+    "mParameters":{
+      "fileIds":[
+        1,
+        2
+      ]
+    }
+  }
+}
```
Comment (yupeng9) [inline servers/src/test/resources/JournalEntries.json]: Please add a new line.

File: servers/src/test/java/tachyon/master/journal/JsonJournalFormatterTest.java [hunk 4]
```diff
@@ -15,9 +15,47 @@
...
+
+  @BeforeClass
+  public static void beforeClass() throws IOException {
+    String entriesFile = JsonJournalFormatterTest.class.getResource(JSON_SAMPLE_PATH).getFile();
+    sRootNode = new ObjectMapper().readTree(new File(entriesFile));
+  }
+
+  @Test
+  public void entriesJsonTest() throws IOException {
+
```
Comment (yupeng9) [inline servers/src/test/java/tachyon/master/journal/JsonJournalFormatterTest.java]: extra blank line

**Comment by yupeng9** [comment]
Thanks for the improvement. I left some minor comments on style.

**Comment by yupeng9** [comment]
LGTM +1

File: servers/src/test/java/tachyon/master/journal/JournalFormatterTestBase.java [hunk 5]
```diff
@@ -87,6 +154,7 @@
 
   @Before
   public final void before() throws Exception {
+
```
Comment (gpang) [inline servers/src/test/java/tachyon/master/journal/JournalFormatterTestBase.java]: extra blank line

File: servers/src/test/java/tachyon/master/journal/JournalFormatterTestBase.java [hunk 6]
```diff
@@ -94,14 +162,12 @@ public final void before() throws Exception {
 
   protected void write(JournalEntry entry) throws IOException {
     mFormatter.serialize(new SerializableJournalEntry(TEST_SEQUENCE_NUMBER, entry), mOs);
-    mOs.close();
```
Comment (gpang) [inline servers/src/test/java/tachyon/master/journal/JournalFormatterTestBase.java]: Why is this removed?

File: servers/src/test/java/tachyon/master/journal/JournalFormatterTestBase.java [hunk 7]
```diff
@@ -94,14 +162,12 @@ public final void before() throws Exception {
...
   }
 
   protected JournalEntry read() throws IOException {
     JournalInputStream jis = mFormatter.deserialize(mIs);
     JournalEntry entry = jis.getNextEntry();
     Assert.assertEquals(TEST_SEQUENCE_NUMBER, jis.getLatestSequenceNumber());
-    jis.close();
```
Comment (gpang) [inline servers/src/test/java/tachyon/master/journal/JournalFormatterTestBase.java]: why is this removed?

File: servers/src/test/resources/JournalEntries.json [hunk 8]
```diff
@@ -0,0 +1,248 @@
+{
```
Comment (gpang) [inline servers/src/test/resources/JournalEntries.json]: I think this file should be called "JsonJournalEntries.json" or something, since this is only used for the "JsonJournalFormatterTest".

**Comment by gpang** [comment]
Thanks for the PR! I just left a few comments.

**Comment by saveveltri** [comment]
Thanks for your suggestions.
The 2 close statements were deleted because their enclosing methods (read and write) are called multiple times during the test execution, i.e. multiple read and write operations are performed on those streams.
In the former implementation, every test method perform only one read and one write operation, so the fact that the stream were closed had no effect while with my implementation, it cause a "stream closed" exception.
Your suggestion made me notice that i forgot to put the close statement in a method with after annotation.

**Comment by uronce-cc** [comment]
+1 LGTM

**Comment by gpang** [comment]
Thanks for the explanations!

**Comment by gpang** [comment]
LGTM
